schematic/utils/google_api_utils.py

```diff
@@ -23,55 +17,62 @@
+    # The file token.pickle stores the user's access and refresh tokens,
     # and is created automatically when the authorization flow completes for the first time.
-    if os.path.exists(TOKEN_PICKLE):
-        with open(TOKEN_PICKLE, 'rb') as token:
+    if os.path.exists(CONFIG.TOKEN_PICKLE):
+        with open(CONFIG.TOKEN_PICKLE, 'rb') as token:
             creds = pickle.load(token)
 
     # If there are no (valid) credentials available, let the user log in.
     if not creds or not creds.valid:
         if creds and creds.expired and creds.refresh_token:
             creds.refresh(Request())
         else:
-            flow = InstalledAppFlow.from_client_secrets_file(CREDS_PATH, SCOPES)
+            flow = InstalledAppFlow.from_client_secrets_file(CONFIG.CREDS_PATH, SCOPES)
             creds = flow.run_console() ### don't have to deal with ports
         # Save the credentials for the next run
-        with open(TOKEN_PICKLE, 'wb') as token:
+        with open(CONFIG.TOKEN_PICKLE, 'wb') as token:
             pickle.dump(creds, token)
 
     # get a Google Sheet API service
     sheet_service = build('sheets', 'v4', credentials=creds)
     # get a Google Drive API service
     drive_service = build('drive', 'v3', credentials=creds)
-    
+
     return {
         'sheet_service': sheet_service,
         'drive_service': drive_service,
         'creds': creds
     }
 
 def build_service_account_creds():
-    credentials = service_account.Credentials.from_service_account_file(SERVICE_ACCT_CREDS, scopes=SCOPES)
+    credentials = service_account.Credentials.from_service_account_file(CONFIG.SERVICE_ACCT_CREDS, scopes=SCOPES)
 
     # get a Google Sheet API service
     sheet_service = build('sheets', 'v4', credentials=credentials)
     # get a Google Drive API service
     drive_service = build('drive', 'v3', credentials=credentials)
-    
+
     return {
         'sheet_service': sheet_service,
         'drive_service': drive_service,
         'creds': credentials
     }
 
 def download_creds_file():
-    if not os.path.exists(CREDS_PATH):
-    
+    if not os.path.exists(CONFIG.CREDS_PATH):
+
         print("Retrieving Google API credentials from Synapse...")
+        # synapse ID of the 'credentials.json' file, which we need in
+        # order to establish communication with gAPIs/services
+        API_CREDS = CONFIG["synapse"]["api_creds"]
         syn = synapseclient.Synapse()
         syn.login()
-        syn.get(SYN_CREDS, downloadLocation = ROOT_DIR)
+        # Download in parent directory of CREDS_PATH to
+        # ensure same file system for os.rename()
+        creds_dir = os.path.dirname(CONFIG.CREDS_PATH)
+        creds_file = syn.get(API_CREDS, downloadLocation = creds_dir)
+        os.rename(creds_file.path, CONFIG.CREDS_PATH)
         print("Downloaded Google API credentials file.")
 
 def execute_google_api_requests(service, requests_body, **kwargs):
```

Comment:
@sujaypatil96 this is separate from this PR but as we say in Bulgaria, "I left my fingers in the meal" here. This function needs to be better thought out - what kind of parameters we support; what kinds of requests can be executed; what kinds of exceptions can be thrown.